src/sagemaker_containers/environment.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import multiprocessing
+import os
+
+import sys
+
+base_dir = 'opt/ml'
+
+
+class Environment(object):
+    """Provides access to common aspects of the container environment, including
+    important system characteristics, filesystem locations, and configuration settings.
+    """
+
+    def __init__(self):
+        self.base_dir = base_dir
+        sys.path.insert(0, self.code_dir)
+
+    @property
+    def model_dir(self):
+        return os.path.join(self.base_dir, 'model')
```

Comment:
Not a big fan of having the getter function of the property calculate its value every time. Can we move to having:

```python
def __init__(self):
    self._model_dir = os.path.join(self.base_dir, 'model')
     # ... other stuff here

@property
def model_dir(self):
    return self._model_dir
```

If someone else is strongly opposed to this Im fine discussing it though

Reply:
moved all logic to the constructor.